lib/stores/DataStore.js

```diff
@@ -37,13 +41,20 @@ class DataStore {
      *
      * http://tus.io/protocols/resumable-upload.html#creation
      *
-     * @param  {object} file File model.
+     * @param  {object} req http.incomingMessage
+     * @return {Promise}
      */
-    create(file) {
-        if (!(file instanceof File)) {
-            throw new Error(`${file} is not a File`);
-        }
-        console.log(`[DataStore] create: ${file.id}`);
+    create(req) {
+        console.log('[DataStore] create');
+
+        return new Promise((resolve, reject) => {
```

Comment:
Looking at this now, it probably makes sense for a `create` method to resolve with the model its creating. Do you think this should this resolve with the whole `File` rather than just the `file_name`?